alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/test/java/org/alfresco/transformer/AIOControllerLibreOfficeTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * #%L
+ * Alfresco Transform Core
+ * %%
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * -
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ * -
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * -
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.transformer;
+
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.alfresco.transformer.executors.LibreOfficeJavaExecutor;
+import org.alfresco.transformer.transformers.LibreOfficeAdapter;
+import org.alfresco.transformer.transformers.Transformer;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(AIOController.class)
+@Import(AIOCustomConfig.class)
+/**
+ * Test the AIOController without a server.
+ * Super class includes tests for the LibreOfficeController and AbstractTransformerController.
+ */
+public class AIOControllerLibreOfficeTest extends LibreOfficeControllerTest
+{
+
+    //Tests contained in LibreOfficeControllerTest
+    
+    LibreOfficeAdapter adapter;
+
+    @Autowired
+    AIOTransformRegistry transformRegistry;
+
+    @PostConstruct
+    private void init() throws Exception
+    {
+        adapter = new LibreOfficeAdapter(execPath);
+    }
+
+    @Override
+    // Used by the super class to mock the javaExecutor, a different implementation is required here
+    protected void setJavaExecutor(AbstractTransformerController controller, LibreOfficeJavaExecutor javaExecutor)
+    {
+        ReflectionTestUtils.setField(adapter, "javaExecutor", javaExecutor);
+        //Need to wire in the mocked adapter into the controller...
+        if (ReflectionTestUtils.getField(transformRegistry,"transformerTransformMapping") instanceof Map)

```

Comment:
What's the purpose of this if statement? If transformerTransformMapping is not a Map this test should be failing anyway.

Reply:
You can replace the whole thing with just:
Map<String,Transformer> transformers = transformRegistry.getTransformerTransformMapping();
transformers.replace("libreoffice", adapter);
No need to override it with reflection after either.

Reply:
Yes you're right, there was originally some type safety issues in a older version that is no longer present. Removed the additional lines.
Also made the change in ImageMagick